modules/querier/http.go

```diff
@@ -482,3 +471,30 @@ func handleError(w http.ResponseWriter, err error) {
 
 	http.Error(w, err.Error(), http.StatusInternalServerError)
 }
+
+func writeFormattedContentForRequest(w http.ResponseWriter, req *http.Request, m proto.Message) {
+	switch req.Header.Get(api.HeaderAccept) {
+	case api.HeaderAcceptProtobuf:
+		b, err := proto.Marshal(m)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		_, err = w.Write(b)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set(api.HeaderContentType, api.HeaderAcceptProtobuf)
```

Comment:
Good catch. I didn't think about it because the previous code was setting it at the end. It seems there is helpful magic inside Write:  https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/net/http/server.go;l=120   Will move it up, and keep it so that it's explicit and not relying on the content type detection.

```go
//   If the Header
// does not contain a Content-Type line, Write adds a Content-Type set
// to the result of passing the initial 512 bytes of written data to
// [DetectContentType].
```